src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -209,6 +211,16 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
     if (step === IDX_STEP.CANCEL_TRANSACTION) {
       SessionStorage.removeStateHandle();
     }
+    if (step === IDX_STEP.IDENTIFY && features?.deviceFingerprinting) {
+      const baseUrl = getBaseUrl(widgetProps);
+      if (baseUrl) {
+        // Proceeds with form submission even if device fingerprinting fails
+        const fingerprint = await generateDeviceFingerprint(baseUrl).catch(() => undefined);
```

Comment:
Won't this bury the exception/rejection? Should it log before returning `undefined`?

Reply:
Keep it the same as Gen2 for now. TestCafe specs and downstreams may fail if there are unexpected logs/errors in console.